pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1303,6 +1303,46 @@ func parseDeviceIoListConfig(getconfigCtx *getconfigContext,
 					"key: %s, value: %s", key, value)
 			}
 		}
+
+		if ioDevicePtr.Vflist != nil && ioDevicePtr.Vflist.VfCount > 0 {
+			port.Vfs.Count = uint8(ioDevicePtr.Vflist.VfCount)
+			port.Vfs.Data = make([]sriov.EthVF, ioDevicePtr.Vflist.VfCount)
+
+			valid := true
+			for i, vf := range ioDevicePtr.Vflist.Data {
+				// not checking lower bound, since it's zero if VlanId is not specified
+				if vf.VlanId > maxVlanID {
+					log.Errorf("Incorrect VlanID %d for PhysicalIO %s", vf.VlanId, ioDevicePtr)
+					valid = false
+					break
+				}
+
+				port.Vfs.Data[i] = sriov.EthVF{
+					Index:  uint8(vf.Index),
+					Mac:    vf.Mac,
+					VlanID: uint16(vf.VlanId),
+				}
+			}
+			if !valid {
+				continue
+			}
+			// Generate unspecified VFs
+			if len(port.Vfs.Data) != int(ioDevicePtr.Vflist.VfCount) {
+				set := map[int]struct{}{}
+				for _, d := range port.Vfs.Data {
+					set[int(d.Index)] = struct{}{}
+				}
+				for i := 0; i < int(ioDevicePtr.Vflist.VfCount); i++ {
+					if _, ok := set[i]; ok {
+						continue
+					}
+					port.Vfs.Data = append(port.Vfs.Data, sriov.EthVF{
```

Comment:
LGTM, just since data entries for each VF are not necessary, we can just append to port.Vfs.Data in line 1320.

Reply:
I'm not sure I follow. I recall seeing a panic/err in domainmgr if we try to ask for a VF that doesn't exist (which can happen if we set the Count to 8 but only specify 3 VFs in data)? Are you saying that the `Index` field doesn't matter so we can just append empty `sriov.EthVF` structs?

Reply:
What I'm saying is that we can specify 8 VFs without configuring them. And we will create `Vfs.Count` Virtual functions and will configure ones which are specified in `Vfs.Data` field. So I can do this in config
1) Specify `Vfs.Count` for 3 and leave `Vfs.Data` empty -- I don't care about VF configuration I want 3 default VFs from my PF. `Vfs.Data` will be empty in that case
2) Specify `Vfs.Count`for 8 and set only 2 VFs inside `Vfs.Data` -- I really care about configuring VF0 and VF5 rest of them can be default. `Vfs.Data` in that case will be smth like: `[{Index: 0, ....}, {Index: 5, ...}]`
3) Specify `Vfs.Count` for 3 and set them all up. 
So basically rule is within `max(Vfs.Data.Index) < Vfs.Count` ; given than `Vfs.Data.Index` are unique numbers

Reply:
Isn't this what the code is doing? I'm not sure I can append on line 1320 since I think I need to go through the list once to see which indices are specified and only create the ones that are not specified.

Having the index missing in the data has not worked in my tests.